src/recordlinker/routes/person_router.py

```diff
@@ -75,3 +100,58 @@ def update_person(
 
     person = service.update_person_cluster(session, patients, person, commit=False)
     return schemas.PersonRef(person_reference_id=person.reference_id)
+
+
+def persons_by_reference_ids_or_422(
+    session: orm.Session, reference_ids: typing.Sequence[uuid.UUID]
+) -> typing.Sequence[models.Person]:
+    """
+    Retrieve the Persons by their reference ids or raise a 422 error response.
+    """
+    persons = service.get_persons_by_reference_ids(session, reference_ids)
+    if not persons:
+        raise fastapi.HTTPException(
+            status_code=fastapi.status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=[
+                {
+                    "loc": ["body", "persons"],
+                    "msg": "Invalid person_reference_id(s). No persons found for any of the provided person_rederence_id(s)",
+                    "type": "value_error",
+                }
+            ],
+        )
+    return persons  # type: ignore
+
+
+@router.post(
+    "/{person_reference_id}/merge",
+    summary="Merge Person clusters",
+    status_code=fastapi.status.HTTP_200_OK,
+)
+def merge_person_clusters(
+    person_reference_id: uuid.UUID,
+    data: typing.Annotated[schemas.PersonRefs, fastapi.Body()],
+    session: orm.Session = fastapi.Depends(get_session),
+) -> schemas.PersonRef:
+    """
+    Merges Person cluster(s) into the Person cluster referenced by `person_reference_id`.
+    """
+    # Get the person_reference_id the person clusters will be merged into
+    person = service.get_person_by_reference_id(session, person_reference_id)
+
+    if person is None:
+        raise fastapi.HTTPException(status_code=fastapi.status.HTTP_404_NOT_FOUND)
+
+    # Get all persons by person_reference_id that will be merged
+    persons = persons_by_reference_ids_or_422(session, data.person_reference_ids)
+    person_ids = [person.id for person in persons]
+
+    # Get all of the patients from the person clusters to be merged
+    patients = patients_by_person_id_or_422(session, person_ids)
```

Comment:
In what scenario would we raise a 422 on this line?  If I'm reading this correctly, any invalid person_reference_id would be caught on line 146 and result in a raised exception there, so I think the list of person_ids should be safe by the time we get to line 150.

Reply:
Hmm I guess it could depend on how we do clean up of the person clusters. I guess I was thinking of the scenario where patients have been (re) assigned a new person cluster but their empty person cluster is still around...does that make sense? I might not have enough understanding of the current development so I'm open to removing this check if you're comfortable with it.

Reply:
In that case, let's say where a user is trying to merge a person cluster with 0 patients.  I would think we get the empty list back.  Are you thinking that in the case where a user tries to merge cluster B into cluster A, but cluster B is empty that we should return a 422 rather than a noop?